dev/com.ibm.ws.jca.cm/src/com/ibm/ejs/j2c/FreePool.java

```diff
@@ -160,14 +160,26 @@ private void queueRequest(ManagedConnectionFactory managedConnectionFactory, lon
         if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) {
             Tr.entry(this, tc, "queueRequest", waitTimeout);
         }
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled())
+            Tr.debug(tc, "Current connection pool", pm.toString());
         if (waitTimeout == 0) {
             --pm.waiterCount;
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(
+                         tc,
+                         "Timeout.  Decremented waiterCount, which is now "
+                             + pm.waiterCount
+                             + " on datasource "
+                             + gConfigProps.cfName);
+            }
+            Object[] connWaitTimeoutparms = new Object[] { "createOrWaitForConnection", gConfigProps.cfName, 0, pm.waiterCount,
```

Comment:
Should the method here be queueRequest instead of createOrWaitForConnection?

Reply:
done